horton/meanfield/gridgroup.py

```diff
@@ -141,268 +168,293 @@ def compute_energy(self, cache):
         return result
 
     def add_fock(self, cache, *focks):
-        '''Add contributions to the Fock matrix
-
-           **Arguments:**
+        """Add contributions to the Fock matrix.
 
-           cache
-                An instance of Cache, used to store intermediate results.
+        Parameters
+        ----------
+        cache : Cache
+            Used to store intermediate results.
 
-           This method basically dispatches the work to all ``GridObservable``
-           instances in ``self.grid_terms``.
-        '''
+        This method basically dispatches the work to all ``GridObservable`` instances in
+        ``self.grid_terms``.
+        """
         # Get the potentials. If they are not yet evaluated, some computations
         # are needed.
-        dpots, gpots, new = self._get_potentials(cache)
+        lda_pots, gga_pots, mgga_pots, new = self._get_potentials(cache)
 
         if new:
             # compute stuff on the grid that the grid_observables may use
             self._update_grid_data(cache)
 
             # Collect the total potentials.
             for grid_term in self.grid_terms:
-                if grid_term.gga:
-                    grid_term.add_pot(cache, self.grid, *(dpots + gpots))
-                else:
-                    grid_term.add_pot(cache, self.grid, *dpots)
+                if grid_term.df_level == DF_LEVEL_LDA:
```

Comment:
I think Matt is right, this can be done simpler. In the old code that would not have worked, which is the reason for the current code structure. Things can be further simplified now. I'll give it a shot.